### ISSUE TITLE
Lexer cannot deal with delimiters (operators, etc.) touching identifiers

### ISSUE THREAD

**Issue by Matthewar** [Matthewar/ASV#31]
When an operator touches anything but whitespace, an error occurs.
An example of this is the `(3);` in the following ghci log:
```
lexerList "package name is constant name : integer := name (3); end;"
Left Some lexer error occurred
```

**Comment by Matthewar** [comment]
Seems to be that the state `delim` in the lexer is never jumped to, I assume that was supposed to be `identifier`.

**Comment by Matthewar** [comment]
Fix works for test case.
```
lexerList "package name is constant name : integer := name(3); end;"
Right [Keyword Package,Identifier "name",Keyword Is,Keyword Constant,Identifier "name",Operator :,Identifier "integer",Operator :=,Identifier "name",Operator (,Literal (Univ_Int 3),Operator ),Operator ;,Keyword End,Operator ;]
```

**Comment by Matthewar** [comment]
Should do further testing before this is confirmed to be fixed (#3)
